index.js

```diff
@@ -62,15 +61,33 @@ const useNavigate = (options) => {
  * Part 2, Low Carb Router API: Router, Route, Link, Switch
  */
 
-export const Router = (props) => {
-  // this little trick allows to avoid having unnecessary
-  // calls to potentially expensive `buildRouter` method.
+export const Router = ({
+  hook,
+  matcher,
+  base = "",
+  nested = false,
+  children,
+}) => {
+  const parent = useRouter();
+
+  // nested `<Router />` has the scope of its closest parent router (base path is prepended)
+  // Routers are not nested by default, but this might change in future versions
+  const proto = nested ? parent : defaultRouter;
```

Comment:
This change is ok... however... it will lead to inefficiencies down the road. Because you are going to want to change `<Route>` at some point to support nested routes (you can calculate `base` **automatically** from the path regexp!!!)... in which case, you will already have the parent router and can supply it directly to the nested `<Router>` without having to call `useContext()` a second time redundantly. In addition, for non-nested routers, you are now adding an extra hook call for no reason.

My recommendation would be to supply `parent` instead of a `nested` flag. This makes the routers more powerful. And then, turn your focus to allowing users to not have to use Routers at all, by instead allowing nested `Route` to work and auto-calculate the base for the nested router beneath them. ("nested" would be a good flag for Routes instead of Routers!)

Reply:
> This change is ok... however... it will lead to inefficiencies down the road. Because you are going to want to change `<Route>` at some point to support nested routes (you can calculate `base` **automatically** from the path regexp!!!)... in which case, you will already have the parent router and can supply it directly to the nested `<Router>` without having to call `useContext()` a second time redundantly. In addition, for non-nested routers, you are now adding an extra hook call for no reason.
> 
> My recommendation would be to supply `parent` instead of a `nested` flag. This makes the routers more powerful. And then, turn your focus to allowing users to not have to use Routers at all, by instead allowing nested `Route` to work and auto-calculate the base for the nested router beneath them. ("nested" would be a good flag for Routes instead of Routers!)

Not sure if I follow, could you please provide some example on how router and nested routes could be used together?

Reply:
@molefrog Sure, I will put some examples together from my own codebase. Will get back to you soon.

Reply:
Thanks, I'll review the code and get back to you tomorrow.

Reply:
See https://github.com/molefrog/wouter/pull/265/commits/5963014ed7ce69a31240b43d11d71fd188903aae